ci/cscs-ci.yml

```diff
@@ -0,0 +1,91 @@
+include:
+- remote: 'https://gitlab.com/cscs-ci/recipes/-/raw/master/templates/v2/.ci-ext.yml'
+
+.py310: &py310
+  PYVERSION_PREFIX: py310
+  PYVERSION: 3.10.9
+
+.py39: &py39
+  PYVERSION_PREFIX: py39
+  PYVERSION: 3.9.1
+
+.py38: &py38
+  PYVERSION_PREFIX: py38
+  PYVERSION: 3.8.5
+
+stages:
+- baseimage
+- image
+- test
+
+build py38 baseimage:
+  extends: .container-builder
+  stage: baseimage
+  when: manual
+  allow_failure: true
+  variables:
+    PERSIST_IMAGE_NAME: $CSCS_REGISTRY_PATH/base/gt4py-ci:$PYVERSION
+    DOCKERFILE: ci/base.Dockerfile
+    DOCKER_BUILD_ARGS: '["PYVERSION=$PYVERSION", "CI_PROJECT_DIR=$CI_PROJECT_DIR"]'
+    <<: *py38
+
+build py39 baseimage:
+  extends: build py38 baseimage
+  variables:
+    <<: *py39
+
+build py310 baseimage:
+  extends: build py38 baseimage
+  variables:
+    <<: *py310
+
+build py38 image:
+  extends: .container-builder
+  stage: image
+  variables:
+    PERSIST_IMAGE_NAME: $CSCS_REGISTRY_PATH/gt4py/gt4py-ci:$PYVERSION
+    DOCKERFILE: ci/checkout.Dockerfile
+    DOCKER_BUILD_ARGS: '["PYVERSION=$PYVERSION", "CSCS_REGISTRY_PATH=$CSCS_REGISTRY_PATH"]'
+    <<: *py38
+
+build py39 image:
+  extends: build py38 image
+  variables:
+    <<: *py39
+
+build py310 image:
+  extends: build py38 image
+  variables:
+    <<: *py310
+
+test py38:
+  extends: .container-runner-daint-gpu
+  needs: ["build py38 image"]
+  stage: test
+  image: $CSCS_REGISTRY_PATH/gt4py/gt4py-ci:$PYVERSION
+  script:
+  - cd /gt4py.src
+  - python -c "import cupy"
```

Comment:
Another question: is there any value in testing the `cupy` import before running `tox`?

Reply:
If memory serves well, then this is to detect the case where the version of cuda/cupy in the docker image does not match the version of cuda available on daint. I will add a comment to that effect.

Reply:
On second thought I will put that in a separate PR, updating CI docs as well.